### PR TITLE
feat: add support for reverting LV back to state of snapshot 

### PR DESCRIPTION
**Pull request by trgill**
Enhancement: add support for reverting LVs to snapshots

Reason: MVP requirements call for ability to revert snapshots

Use the lvconvert --merge command to revert the changes made to the
LVM logical volume.

If either the source LV or snapshot are open, the merge is deferred
until the next time the server reboots and thesource logical volume
is activated.

Result: Users can revert to snapshots

Issue Tracker Tickets (Jira or BZ if any):

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -35,6 +35,10 @@ This variable is required. It supports one of the following values:
 
 - `clean`: Remove snapshots that conform to the specified prefix and pattern
 
+- `revert`: Revert to snapshots that are specifed by either the pattern or set.  If either the source LV or
+            snapshot are open, the merge is deferred until the next time the server reboots and the
+            source logical volume is activated.
```
Comment (richm) [inline README.md:40]: Is it common for users to want to reboot the system after doing this?
Will users know that the snapshot is open, and that a reboot is required for the merge to apply?

We have other roles that require a reboot for the changes to take effect.
https://github.com/linux-system-roles/kernel_settings#kernel_settings_reboot_ok
and
https://github.com/linux-system-roles/kernel_settings#variables-exported-by-the-role

basically, if the role needs to reboot for a change to take effect, if you set `$ROLENAME_reboot_ok: true`, then the role will reboot the machine.  Otherwise, the role will set `$ROLENAME_reboot_required` to `true` to let the user know that a reboot is required for the changes to take effect.

Comment (trgill) [reply README.md:40]: > Is it common for users to want to reboot the system after doing this? Will users know that the snapshot is open, and that a reboot is required for the merge to apply?

Yes, I think it is common to want a reboot.  But, some users might want to avoid a reboot.

If users are merging/reverting to LVM snapshots, I think they would know.  We should reiterate the rules in the role documentation.

> 
> We have other roles that require a reboot for the changes to take effect. https://github.com/linux-system-roles/kernel_settings#kernel_settings_reboot_ok and https://github.com/linux-system-roles/kernel_settings#variables-exported-by-the-role
> 
> basically, if the role needs to reboot for a change to take effect, if you set `$ROLENAME_reboot_ok: true`, then the role will reboot the machine. Otherwise, the role will set `$ROLENAME_reboot_required` to `true` to let the user know that a reboot is required for the changes to take effect.

I opened an issue to track : https://github.com/linux-system-roles/snapshot/issues/16

File: tasks/files/snapshot.py [hunk 2]
```diff
@@ -302,6 +308,75 @@ def lvm_snapshot_remove(vg_name, snapshot_name):
...
+                SnapshotStatus.ERROR_REVERT_FAILED,
+                "LV with name: " + vg_name + "/" + snapshot_name + " is not a snapshot",
+            )
+    else:
+        return (
+            SnapshotStatus.ERROR_REVERT_FAILED,
+            "snapshot not found with name: " + vg_name + "/" + snapshot_name,
+        )
+
+    revert_command = ["lvconvert", "--merge", vg_name + "/" + snapshot_name]
```
[the remark this answers is not in the document]
Comment (trgill) [reply tasks/files/snapshot.py:328]: Yes, we can detect is a filesystem is mounted on the underlying block device or if an application has the block device open.

Ok if I add that later and we track via the issue?

**Comment by richm** [comment]
[citest]

**Review by richm — APPROVED** [review]
